### ISSUE TITLE
Scour should only change precision if -p is given

### ISSUE THREAD

**Issue by launchpad-issue-importer** [scour-project/scour#80]
Currenly, Scour by default drops precision to five digits, so a file using 7 digits of precision (not all uncommon for Adobe Illustrator files, especially dumb ones, where the whole image is in a view box somewhere 7200 units away from the coordinate system origin) will silently be degraded, without warning.

At least I would not expect lossy behaviour by default; it's non-unix:y.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2011-06-07T00:37:35Z
 * owner: ecmanaut
 * the launchpad url was https://bugs.launchpad.net/bugs/793811

**Comment by launchpad-issue-importer** [comment]
(by ecmanaut)
Along the same lines, it might be good to break up "Number of bytes saved in path data: X" into "Number of bytes saved in path optimization: Y" and "Number of bytes saved in reduced path precision: Z", while at it, as the current number X is a mix of good and maybe-bad trimming.

**Comment by launchpad-issue-importer** [comment]
(by ecmanaut)
Submitting a test case that gets severely munged by the precision loss.

**Comment by Eitot** [comment]
The test case can be found here: https://launchpadlibrarian.net/73124614/a.svg.

The result can definitely be seen when the zoom factor is increased significantly, which is of course a possibility with vector graphics. Ideally, Scour should indeed not presuppose and reduce the precision of significant digits. But, making the change now subverts expected behaviour that has been there for years. Path-heavy SVG files could lose quite a bit of their size reduction without this feature. Changing it should be relatively easy, the default value just has to be a lot bigger, although it doesn’t seem to be possible to simply impose no limit at all. Decimal uses 28 significant digits [by default](https://docs.python.org/2/library/decimal.html).

**Comment by Ede123** [comment]
I don't think one problematic sample justifies a change of the default value here. After all precision can be easily changed if required for very specific edge cases. If you look at the path data it's apparent that the visual difference is a result of a very unfortunate mix of relative and absolute coordinates that every halfway decent SVG editor would have avoided.

I used Scour a lot and the default precision was always more than enough. After all we're talking about 5 significant digits! That's a lot and the error introduced should typically be tiny and mostly invisible even at larger sizes.

In this case the only question is if we can optimize the parsing of path data to avoid this problem altogether... If all processing is done with full precision and the number of significant digits is reduced as the last operation this should be avoidable.

**Comment by Eitot** [comment]
Suppose that this problem only really becomes noticeable when a small vector is enlarged, perhaps it is possible to adapt the precision based upon size of the canvas and the view port. Small vectors may benefit more from a higher precision. But this requires extensive testing to discover the threshold.

**Comment by Ede123** [comment]
I just created #108 which completely fixes the glitch in path data without the need for any change in precision.

While [the actual fix](9e4b9d6f5ed680d8c269c97641fae5ad68c5c9c2) was only a few lines of code it involves some profound changes in processing and I also made some further adjustments because of issues that got apparent as a result of the increased accuracy of processed path data.

Therefore I'd like to encourage you to test the PR for any regressions I might not have cached.
